internal/etw/eventopt.go

```diff
@@ -12,23 +23,35 @@ func WithEventOpts(opts ...EventOpt) []EventOpt {
 
 // WithLevel specifies the level of the event to be written.
 func WithLevel(level Level) EventOpt {
-	return func(descriptor *EventDescriptor, tags *uint32) {
-		descriptor.Level = level
+	return func(options *eventOptions) {
+		options.descriptor.Level = level
 	}
 }
 
 // WithKeyword specifies the keywords of the event to be written. Multiple uses
 // of this option are OR'd together.
 func WithKeyword(keyword uint64) EventOpt {
-	return func(descriptor *EventDescriptor, tags *uint32) {
-		descriptor.Keyword |= keyword
+	return func(options *eventOptions) {
+		options.descriptor.Keyword |= keyword
 	}
 }
 
 // WithTags specifies the tags of the event to be written. Tags is a 28-bit
 // value (top 4 bits are ignored) which are interpreted by the event consumer.
 func WithTags(newTags uint32) EventOpt {
-	return func(descriptor *EventDescriptor, tags *uint32) {
-		*tags |= newTags
+	return func(options *eventOptions) {
+		options.tags |= newTags
+	}
+}
+
+func WithActivityID(activityID *windows.GUID) EventOpt {

```

Comment:
Does this have to be a GUID or can it be any field?

Reply:
Activity ID must be a GUID, we pass it to the LPCGUID parameter on EventWriteTransfer: https://docs.microsoft.com/en-us/windows/desktop/api/evntprov/nf-evntprov-eventwritetransfer